File: SymbolStore/SymbolCache.cs
using System;
using System.IO;

namespace SourceLink.SymbolStore
{
    public class SymbolCache
    {
        string symbolCacheDir;
        IntPtr sessionCookie;

        public SymbolCache(string symbolCacheDir)
        {
            this.symbolCacheDir = symbolCacheDir;
            this.sessionCookie = new IntPtr(new Random().Next());
            SrcSrv.Init(sessionCookie, symbolCacheDir);
        }

        public PdbReader ReadPdb(string filePath, Stream stream)
        {
            return new PdbReader(stream, sessionCookie, filePath);
        }

        public string DownloadFile(string downloadUrl)
        {
            return SymSrv.DownloadFile(downloadUrl, this.symbolCacheDir);
        }
    }
}

File: SymbolStore/SymbolCache.fs
namespace SourceLink.SymbolStore

open System
open System.IO
open SourceLink.SymbolStore.CorSym

type SymbolCache (symbolCacheDir:string) =
    let sessionCookie = IntPtr(Random().Next())
    do SrcSrv.Init(sessionCookie, symbolCacheDir)

    member x.ReadPdb (stream:Stream) filePath =
        PdbReader(stream, sessionCookie, filePath)

    member x.DownloadFile downloadUrl =
        SymSrv.DownloadFile(downloadUrl, symbolCacheDir)